sql/src/main/java/org/apache/apex/malhar/sql/table/FileEndpoint.java

```diff
@@ -0,0 +1,114 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.sql.table;
+
+import java.util.Map;
+
+import org.apache.apex.malhar.sql.operators.LineReader;
+import org.apache.apex.malhar.sql.operators.LineWriter;
+import org.apache.apex.malhar.sql.operators.OperatorUtils;
+import org.apache.apex.malhar.sql.planner.RelInfo;
+import org.apache.calcite.adapter.java.JavaTypeFactory;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.Operator;
+
+/**
+ * This is an implementation of {@link Endpoint} which defined how data should be read/written to file system.
+ */
+public class FileEndpoint implements Endpoint
+{
+  public static final String FILE_INPUT_DIRECTORY = "directory";
+  public static final String FILE_OUT_PATH = "outputFilePath";
+  public static final String FILE_OUT_NAME = "outputFileName";
+
+  private MessageFormat messageFormat;
+
+  private Map<String, Object> operands;
+
+  public FileEndpoint()
+  {
+  }
+
+  public FileEndpoint(String directory, MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+    this.operands = ImmutableMap.<String, Object>of(FILE_INPUT_DIRECTORY, directory);
+  }
+
+  public FileEndpoint(String directory, String fileName, MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+    this.operands = ImmutableMap.<String, Object>of(FILE_OUT_PATH, directory, FILE_OUT_NAME, fileName);
+  }
+
+  @Override
+  public EndpointType getTargetType()
+  {
+    return EndpointType.FILE;
+  }
+
+  @Override
+  public void setEndpointOperands(Map<String, Object> operands)
+  {
+    this.operands = operands;
+  }
+
+  @Override
+  public void setMessageFormat(MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+  }
+
+  @Override
+  public RelInfo populateInputDAG(DAG dag, JavaTypeFactory typeFactory)
+  {
+    LineReader fileInput = dag.addOperator(OperatorUtils.getUniqueOperatorName("FileInput"), LineReader.class);
+    fileInput.setDirectory((String)operands.get(FILE_INPUT_DIRECTORY));
```

Comment:
Created a Jira ticket for this: https://issues.apache.org/jira/browse/APEXMALHAR-2311

@yogidevendra That's a nice suggestion. Can you opinion about this on the Jira? Thanks.